src/nvector/CMakeLists.txt

```diff
@@ -73,8 +73,13 @@ if(BUILD_NVECTOR_KOKKOS)
   install(CODE "MESSAGE(\"\nInstall NVECTOR_KOKKOS\n\")")
   message(STATUS "Added BUILD_NVECTOR_KOKKOS module")
   add_library(sundials_nveckokkos INTERFACE)
-  target_link_libraries(sundials_nveckokkos INTERFACE
+  if(BUILD_NVECTOR_HIP)
```

Comment:
It's possible to configure a build with Kokkos+hip but without enabling the HIP vector. To prevent this we should add a check to `cmake/tpl/SundialsKokkos.cmake` that throws an error if HIP is not enabled as well when Kokkos has HIP support (and similarly CUDA).

Reply:
I changed it to check if HIP is enabled for now as it seems the `Kokkos_DEVICE` cmake variable is unreliable at this time (according to the comment in `cmake/tpl/SundialsKokkos.cmake`).